src/cursor/cur_layered.c

```diff
@@ -666,6 +673,78 @@ __clayered_reset(WT_CURSOR *cursor)
     API_END_RET(session, ret);
 }
 
+/*
+ * __clayered_copy_constituent_bound --
+ *     Copy the top level bound into a single constituent cursor
+ */
+static int
+__clayered_copy_constituent_bound(WT_CURSOR_LAYERED *clayered, WT_CURSOR *constituent)
+{
+    WT_CURSOR *base_cursor;
+    WT_SESSION_IMPL *session;
+
+    session = CUR2S(clayered);
+    base_cursor = (WT_CURSOR *)clayered;
+
+    if (constituent == NULL)
+        return (0);
+
+    /* Note that the inclusive flag is additive to upper/lower, so no need to check it as well */
+    if (F_ISSET(base_cursor, WT_CURSTD_BOUND_UPPER))
+        WT_RET(__wt_buf_dup(session, &constituent->upper_bound, &base_cursor->upper_bound));
+    else {
+        __wt_buf_free(session, &constituent->upper_bound);
+        WT_CLEAR(constituent->upper_bound);
+    }
+    if (F_ISSET(base_cursor, WT_CURSTD_BOUND_LOWER))
+        WT_RET(__wt_buf_dup(session, &constituent->lower_bound, &base_cursor->lower_bound));
+    else {
+        __wt_buf_free(session, &constituent->lower_bound);
+        WT_CLEAR(constituent->lower_bound);
+    }
+    /* Copy across all the bound configurations */
+    F_SET(constituent, F_MASK(base_cursor, WT_CURSTD_BOUND_ALL));
```

Comment:
Is this correct? If we want to clear the flag, it seems not work? F_SET is or operation.

Reply:
Thanks! I fixed this by using the pattern that's used [here|https://github.com/wiredtiger/wiredtiger/blob/develop/src/btree/bt_cursor.c#L117]. Will push it up soon